public/src/components/defaultPromos.tsx

```diff
@@ -0,0 +1,94 @@
+import { Button, TextField, makeStyles } from '@material-ui/core';
+import React, { useState } from 'react';
+import withS3Data, { InnerProps } from '../hocs/withS3Data';
+import { parsePromoInput } from '../utils/parsePromoInput';
+import {
+  fetchSupportFrontendSettings,
+  saveSupportFrontendSettings,
+  SupportFrontendSettingsType,
+} from '../utils/requests';
+
+type ProductName = 'guardianWeekly' | 'paper';
+
+type DefaultPromos = {
+  [key in ProductName]: string[];
+};
+
+const useStyles = makeStyles(() => ({
+  container: {
+    margin: '30px',
+    maxWidth: '500px',
+    display: 'flex',
+    flexDirection: 'column',
+    '& > * + *': {
+      marginTop: '5px',
+    },
+  },
+}));
+
+const DefaultPromos: React.FC<InnerProps<DefaultPromos>> = ({
+  data,
+  setData,
+  saveData,
+  saving,
+}: InnerProps<DefaultPromos>) => {
+  const [gwPromosString, setGwPromosString] = useState<string>(data.guardianWeekly.join(', '));
+  const [paperPromosString, setpaperPromosString] = useState<string>(data.paper.join(', '));
```

Comment:
Are these `useState`s actually needed?
Because you're calling `setData` in the `onChange` handlers I think this would have the same effect -
```
<TextField
  value={data.guardianWeekly.join(', ')}
  ...
/>
```

Reply:
We started out with exactly that, but found interacting with the text field felt quite awkward (as it was constantly reformatted as you type, for example superfluous spaces being immediately removed).

Reply:
actually I'm wrong, this wouldn't work!

Reply:
ah, your reply didn't appear until I sent the last message! This makes sense.
I guess the `setData` call could happen onBlur, but it probably doesn't make much difference